### ISSUE TITLE
Following build/test instructions doesn't build any tests

### ISSUE THREAD

**Issue by vellozzi** [dotnet/coreclr#17924]
I'm trying to build coreclr release/2.1 on Amazon Linux 2017.03 and can't get the tests to build.  (I have the same issue on Ubuntu)
Here are the instructions I'm following:
https://github.com/dotnet/coreclr/blob/release/2.1/Documentation/building/linux-instructions.md
https://github.com/dotnet/coreclr/blob/release/2.1/Documentation/building/unix-test-instructions.md

Here's my script:
```
# build coreclr
mkdir -p ~/build
cd ~/build
git clone https://github.com/dotnet/coreclr.git
cd coreclr
git checkout release/2.1

./build.sh x64 release msbuildonunsupportedplatform
./build-test.sh x64 release msbuildonunsupportedplatform

# build corefx
cd ~/build
git clone https://github.com/dotnet/corefx.git
cd corefx
git checkout release/2.1

./build.sh -release -RuntimeOS=linux

# test coreclr
cd ~/build/coreclr

tests/runtest.sh \
    --testRootDir=/home/user/build/coreclr/bin/tests/Linux.x64.Release \
    --testNativeBinDir=/home/user/build/coreclr/bin/obj/Linux.x64.Release/tests \
    --coreClrBinDir=/home/user/build/coreclr/bin/Product/Linux.x64.Release \
    --mscorlibDir=/home/user/build/coreclr/bin/Product/Linux.x64.Release \
    --coreFxBinDir=/home/user/build/corefx/bin/runtime/netcoreapp-Linux-Release-x64
```

This results in:
```
.
.
.
The tests have been prepared

=======================
     Test Results
=======================
# CoreCLR Bin Dir  : /home/user/build/coreclr/bin/Product/Linux.x64.Release
# Tests Discovered : 0
# Passed           : 0
# Failed           : 0
# Skipped          : 0
=======================
0 minutes and 0 seconds taken to run CoreCLR tests.
```
The main problem is that build-test.sh isn't building the tests.  If I build an individual test:
```
# from https://github.com/dotnet/coreclr/blob/release/2.1/Documentation/building/unix-test-instructions.md
coreclr/Tools/msbuild.sh /maxcpucount coreclr/tests/src/JIT/CodeGenBringUpTests/Array1.csproj /p:__BuildType=Release /p:__BuildOS=Linux
```
Then when I run my runtest.sh command from above the test is run.

How do I get the tests to build?

**Comment by 4creators** [comment]
The Linux test build was broken for short time and was fixed by #17725 PR but the fix was applied only to master branch.

@jashook @BruceForstall Should I port the fix to 2.1 branch as well?

**Comment by 4creators** [comment]
In case #17725 would be ported to 2.1 it would be helpful to port #17785 as well

**Comment by jashook** [comment]
@RussKeldorph

**Comment by vellozzi** [comment]
Thanks for the update.  Please do port those PRs to the 2.1 branch.

**Comment by jashook** [comment]
Will handle porting these to 2.1.

**Comment by jashook** [comment]
Fixed with #17965
